### PR TITLE
Adopt memorystore for tests

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
This PR adopts the MemoryStore (https://github.com/mattermost/mattermost/pull/26244) from the tests in `server/plugin/subscriptions_test.go` and removes the mocks.

I've included a cleanup of `p.API` usage via 3f0f4fc4d0f33e8d57b99301a0df078286d30663 in this PR.

#### Ticket Link
None

### CODE REVIEW

**Comment by hanzei** [comment]
@mickmister @raghavaggarwal2308 Gentle reminder to review the PR

**Review by raghavaggarwal2308 — APPROVED** [review]
@hanzei LGTM 👍🏽 

Note: tests are failing in CI

**Comment by hanzei** [comment]
I've reached out to @ayusht2810 regarding the failing CI in https://community.mattermost.com/core/pl/z7833z1f6iyaibmib4af13fzar

File: server/plugin/subscriptions_test.go [hunk 1]
```diff
@@ -58,7 +59,7 @@ func TestPlugin_GetSubscriptionsByChannel(t *testing.T) {
 		{
 			name: "basic test",
 			args: args{channelID: "1"},
-			plugin: pluginWithMockedSubs([]*Subscription{
+			plugin: pluginWithSubs(t, []*Subscription{
```
Comment (mickmister) [inline server/plugin/subscriptions_test.go:62]: Nice function name change :+1:

**Review by mickmister — APPROVED** [review]

**Comment by raghavaggarwal2308** [comment]
> I've reached out to @ayusht2810 regarding the failing CI in https://community.mattermost.com/core/pl/z7833z1f6iyaibmib4af13fzar

@hanzei The tests are working fine on my local for master and your branch as well. Can you please try re-running the CI here?

**Comment by hanzei** [comment]
@raghavaggarwal2308 I've re-triggered CI, which fixed the issue.

Please note that the tests on `master` also fail. It seems like we have a flaky test there.

**Comment by hanzei** [comment]
@raghavaggarwal2308 @mickmister Do you think adopting the memory store would be helpful for other plugins that extensively make use of the KV store like Gitlab or JIRA?

**Comment by hanzei** [comment]
@raghavaggarwal2308 The nightly run failed again: https://github.com/mattermost/mattermost-plugin-github/actions/runs/8164766247/job/22320677668

**Comment by Kshitij-Katiyar** [comment]
> @raghavaggarwal2308 The nightly run failed again: https://github.com/mattermost/mattermost-plugin-github/actions/runs/8164766247/job/22320677668

@hanzei I tried running the commands locally many times but was not really able to reproduce this. 
Screenshot from 2024-03-19 18-49-21

**Comment by hanzei** [comment]
My own tests were the ones that were flaky. :facepalm: Submitted https://github.com/mattermost/mattermost-plugin-starter-template/pull/197.

**Pull request merged**